### PR TITLE
Pramaterize the existing pachyderm-bootstrap-config k8s secret

### PR DESCRIPTION
**Pull request by acohen4**

### CODE REVIEW

File: etc/helm/examples/local-dev-values.yaml [hunk 1]
```diff
@@ -16,10 +16,22 @@ pachd:
   clusterDeploymentID: dev
   enterpriseLicenseKey: ""
 
+  oidc:
+    mockIDP: false
+    clientId: "pachd"
+    issuer: "http://pachd:1658/"
```
Comment (tybritten) [inline etc/helm/examples/local-dev-values.yaml:22]:
```suggestion
    issuer: "http://pachd:30658/"
```

**Review by tybritten — CHANGES_REQUESTED** [review]
LGTM besides those port changes

**Comment by acohen4** [comment]
> LGTM besides those port changes

Ah right. So this is kinda weird but the reason pachd:1658 is used for LOCAL deploys is because in local deploys, we set the pachd k8s service to type "NodePort", which then forces us to use the in-cluster service address port as 1658 instead of 30658. 
We do that here: https://github.com/pachyderm/pachyderm/blob/master/etc/helm/pachyderm/templates/pachd/service.yaml#L40

I'm not sure what the best way to communicate it is for folks doing iterative deployments where they initially want to setup localhost in their OAuth redirect URI. One option could be adding values with pachd:30658 to a non-local helm values.yaml.

What do you think @tybritten ?

**Comment by tybritten** [comment]
if we're doing nodeport and its 30658 (https://github.com/pachyderm/pachyderm/blob/master/etc/helm/pachyderm/templates/pachd/service.yaml#L41) why would't that work for local?

**Comment by acohen4** [comment]
> if we're doing nodeport and its 30658 (https://github.com/pachyderm/pachyderm/blob/master/etc/helm/pachyderm/templates/pachd/service.yaml#L41) why would't that work for local?

because Console -> pachd can't go through Nodeport i.e. `localhost:30658`.  It would instead have to go through `pachd:1658`, and since the Issuer URL needs to be the same value on side of the issuer as on the side of the client, local pachd's identity service must also see itself as `pachd:1658`.

Related PR: https://github.com/pachyderm/pachyderm/pull/6548

**Comment by tybritten** [comment]
Ah, can't we set both nodeport and port to 30658 there?

**Comment by acohen4** [comment]
> Ah, can't we set both nodeport and port to 30658 there?

Good point! That's a viable (and simpler) alternative for that. 

I'm not sure what the exact reason is that we are careful to to have `nodeport != port`. That's something @chainlink could weigh in on when he's back, or perhaps @robert-uhl knows.

File: etc/helm/examples/local-values.yaml [hunk 2]
```diff
@@ -3,12 +3,39 @@
 
 deployTarget: LOCAL
 
+pachd:
+  # registerWithEnterpriseOnly: false
```
Comment (acohen4) [inline etc/helm/examples/local-values.yaml:7]: I'll delete this line

**Comment by chainlink** [comment]
Can we access the nodeport from inside the cluster? (I don't think so, but am unsure)

**Comment by tybritten** [comment]
no, but you can set both the nodeport and port to the same number and inside the cluster will hit `pachd:30658` and the local os outside the cluster will hit the nodeport at 20658

File: etc/helm/pachyderm/values.yaml [hunk 3]
```diff
@@ -212,7 +212,18 @@ pachd:
   enterpriseLicenseKey: ""
   rootToken: ""
   enterpriseSecret: ""
-  oauthClientSecret: ""
+  oidc:
+    mockIDP: false
+    clientId: ""
```
Comment (chainlink) [inline etc/helm/pachyderm/values.yaml:217]: Set default to `pachd` to replicate prior behavior

**Comment by acohen4** [comment]
@tybritten I'd like to handle the Port == NodePort change outside of this PR.

 I made an Issue to track this: https://github.com/pachyderm/pachyderm/issues/6901
I would also like to do one last round of thoughts on whether this change is actually desirable. My main concern is whether this would be unconventional, and if it is in fact unconventional, is it worth doing for a simpler configuration?

**Review by chainlink — CHANGES_REQUESTED** [review]
Looking good! A couple of comments:
- Removing `pachd.oauthClientSecret` will technically break compatibility, we may want to consider having the new variables follow the old convention (ie. `oauthClientName`) instead of moving to a new stanza.
- idps / connectors should just be passed through as a `toYaml` (with the exception of the special case for mockIDP)

File: etc/helm/examples/local-dev-values.yaml [hunk 4]
```diff
@@ -15,11 +15,21 @@ pachd:
...
+    issuer: ""
+    clientID: ""
+    clientSecret: ""
+    redirectURI: "http://localhost:30658/callback"
 
 console:
   enabled: false
   config:
-    issuerURI: "http://pachd:1658"
+    issuerURI: "http://pachd:1658/"
```
Comment (chainlink) [inline etc/helm/examples/local-dev-values.yaml:32]: Is a trailing slash always required here?

File: etc/helm/examples/local-values.yaml [hunk 5]
```diff
@@ -3,12 +3,37 @@
...
     oauthClientID: "console"
     oauthClientSecret: "abc"
     graphqlPort: 4000
     oauthPachdClientID: "pachd"
     pachdAddress: "pachd-peer.default.svc.cluster.local:30653"
+
+enterpriseServer:
+  enabled: false
+
+etcd:
```
[the remark this answers is not in the document]
Comment (acohen4) [reply etc/helm/examples/local-values.yaml:33]: will remove this

File: etc/helm/examples/local-values.yaml [hunk 6]
```diff
@@ -3,12 +3,37 @@
 
 deployTarget: LOCAL
 
+pachd:
+  enterpriseLicenseKey: "license"
```
[the remark this answers is not in the document]
Comment (acohen4) [reply etc/helm/examples/local-values.yaml:7]: will remove this

File: etc/helm/examples/local-dev-values.yaml (hunk 4 above)
Comment (acohen4) [reply etc/helm/examples/local-dev-values.yaml:32]: verified that the backslash isn't essential. The issuer URI value just needs to be consistent between console and pachd

**Review by chainlink — APPROVED** [review]

**Pull request merged**